src/cmd-buildextend-metal

```diff
@@ -118,26 +113,43 @@ if [ "${rev_parsed}" != "${commit}" ]; then
     fi # otherwise, the repo already has a ref, so no need to create
 fi
 
+
+itype=metal
+img=${name}-${build}-${itype}.raw
+if [ -f "${builddir}/${img}" ]; then
+    echo "Bare metal image already exists"
+    exit
+fi
+
+path=${PWD}/${img}
 # for anaconda logs
 mkdir -p tmp/anaconda
 
-for itype in ${BIOS:+metal-bios} ${UEFI:+metal-uefi}; do
-    img=${name}-${build}-${itype}.raw
-    if [ -f "${builddir}/${img}" ]; then
-        echo "Image $itype already exists"
-        continue
-    fi
-
-    path=${PWD}/${img}
-    run_virtinstall "${ostree_repo}" "${ref}" "${path}.tmp" --variant="${itype}"
+if [ "$arch" == "aarch64" ]; then
+    run_virtinstall "${ostree_repo}" "${ref}" "${path}.tmp" --variant="${itype}-uefi"
     /usr/lib/coreos-assembler/gf-platformid "${path}"{.tmp,} metal
     rm -f "${path}".tmp
+else
+    if [ ! -e "$PWD/tmp/ostree-size.json" ]; then
+        /usr/lib/coreos-assembler/estimate-commit-disk-size --repo "$ostree_repo" "$ref" > "$PWD/tmp/ostree-size.json"
+    fi
+    size="$(jq '."estimate-mb".final' "$PWD/tmp/ostree-size.json")"
+    # 384M is the non-ostree partitions
```

Comment:
Random thought related to this: I remember we recently discussed the fact that because we don't actually expand until the real root, it technically limits how much Ignition can write to the disk. Or I forgot, did we say we want to switch that auto-expand logic to a base config so `ignition-disks` does it?

Reply:
I don't remember where we stand on that either, but it's somewhat disconnected from this PR. Ignition disks notably can't extend filesystems, only partitions (fixing that is... complicated). We might think about doing something between disks and files though (effectively moving `coreos-growpart` or something like it to the initramfs).

This currently matches the existing behavior, lets separate any changes there to another PR?

Reply:
Yup, WFM!